File: app/src/main/java/ru/goodibunakov/prodvtest/utils/DateUtils.java
package ru.goodibunakov.prodvtest.utils;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class DateUtils {

    private static final String SERVER_DATE_FORMAT = "yyyy-MM-dd HH:mm:ss";
    private static final String UI_DATE_FORMAT = "dd MMM";
    private static final String DATE_FORMAT = "yyyy-MM-dd";

    public static String convertDateForUI(String dateIn){
        SimpleDateFormat inputFormat = new SimpleDateFormat(SERVER_DATE_FORMAT, Locale.getDefault());
        SimpleDateFormat outputFormat = new SimpleDateFormat(UI_DATE_FORMAT, Locale.getDefault());
        Date date = null;
        try {
            date = inputFormat.parse(dateIn);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        if (date != null){
            return outputFormat.format(date);
        } else {
            return "";
        }

    }

    public static String convertDate(String dateIn){
        SimpleDateFormat inputFormat = new SimpleDateFormat(SERVER_DATE_FORMAT, Locale.getDefault());
        SimpleDateFormat outputFormat = new SimpleDateFormat(DATE_FORMAT, Locale.getDefault());
        Date date = null;
        try {
            date = inputFormat.parse(dateIn);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        if (date != null){
            return outputFormat.format(date);
        } else {
            return "";
        }

    }
}

File: app/src/main/java/ru/goodibunakov/prodvtest/utils/DateUtils.kt
package ru.goodibunakov.prodvtest.utils

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

object DateUtils {

    private const val SERVER_DATE_FORMAT = "yyyy-MM-dd HH:mm:ss"
    private const val UI_DATE_FORMAT = "dd MMM"
    private const val DATE_FORMAT = "yyyy-MM-dd"

    fun convertDateForUI(dateIn: String): String {
        val inputFormat = SimpleDateFormat(SERVER_DATE_FORMAT, Locale.getDefault())
        val outputFormat = SimpleDateFormat(UI_DATE_FORMAT, Locale.getDefault())
        var date: Date? = null
        try {
            date = inputFormat.parse(dateIn)
        } catch (e: ParseException) {
            e.printStackTrace()
        }

        return if (date != null) {
            outputFormat.format(date)
        } else {
            ""
        }
    }

    fun convertDate(dateIn: String): String {
        val inputFormat = SimpleDateFormat(SERVER_DATE_FORMAT, Locale.getDefault())
        val outputFormat = SimpleDateFormat(DATE_FORMAT, Locale.getDefault())
        var date: Date? = null
        try {
            date = inputFormat.parse(dateIn)
        } catch (e: ParseException) {
            e.printStackTrace()
        }

        return if (date != null) {
            outputFormat.format(date)
        } else {
            ""
        }
    }
}